### DISCUSSION TITLE
请问如何在mmyolo当中添加EIoU或者其他IoU的变种呢？

### DISCUSSION THREAD

**Discussion by wannabetter** [open-mmlab/mmyolo#762] [Q&A]
论文当中的源码看起来太费劲了，感觉mmyolo可能对学习有帮助，如题请问怎么给mmyolo添加新IoU或者各种tricks。

**Comment by hhaAndroid** [comment]
在 MMYOLO 中添加其他 IoU 变体的方法一般是修改 `mmyolo/models/losses/iou_loss.py#bbox_overlaps` 中的 `iou_type` 参数和 `bbox_overlaps` 函数实现。其中 `iou_type` 参数可以设置为多个不同的 IoU 变体，例如原始的 IoU 计算、GIoU 计算、DIoU 计算、CIoU 计算等。而 `bbox_overlaps` 函数则实现了这些底层的 IoU 计算逻辑。您可以参考已经实现的 IoU 变体，比如 `bbox_overlaps.py` 中的 `bbox_overlaps` 函数，来实现您想要的 IoU 变体。最后，在您实现的 IoU 计算器上，您可以通过配置文件中的 `bbox_head` 中的 `iou_calculator` 来指定使用哪种计算器进行计算。
